com.reprezen.swagedit.tests/src/com/reprezen/swagedit/validation/ValidatorTest.xtend

```diff
@@ -525,4 +525,93 @@ class ValidatorTest {
 		assertTrue(errors.map[message].forall[it.equals(Messages.error_array_missing_items)])
 		assertThat(errors.map[line], hasItems(12, 17))
 	}
+	
+	@Test
+	def void testValidateMissingTypeInDefinitions() {
+		val content = '''
+		swagger: '2.0'
+		info:
+		  version: 0.0.0
+		  title: Simple API
+		paths:
+		  /foo:
+		    get:
+		      responses:
+		        '200':
+		          description: OK
+		definitions:
+		  Foo:
+		    properties:
+		      bar:
+		        type: string
+		'''
+
+		document.set(content)
+		document.onChange()
+
+		val errors = validator.validate(document, null)		
+		assertEquals(1, errors.size())
+		assertEquals(Messages.error_type_missing, errors.get(0).message)
+	}
+
+	@Test
+	def void testValidateWrongTypeDefinition() {
+		val content = '''
+		swagger: '2.0'
+		info:
+		  version: 0.0.0
+		  title: Simple API
+		paths:
+		  /foo:
+		    get:
+		      responses:
+		        '200':
+		          description: OK
+		definitions:
+		  Foo:
+		    type: string
+		    properties:
+		      bar:
+		        type: string
+		'''
+
+		document.set(content)
+		document.onChange()
+
+		val errors = validator.validate(document, null)		
+		assertEquals(1, errors.size())
+		assertEquals(Messages.error_wrong_type, errors.get(0).message)
+	}
+
+	@Test
+	def void testValidateMissingRequiredProperties() {
+		val content = '''
+		swagger: '2.0'
+		info:
+		  version: 0.0.0
+		  title: Simple API
+		paths:
+		  /foo:
+		    get:
+		      responses:
+		        '200':
+		          description: OK
+		definitions:
+		  Foo:
+		    type: object
+		    properties:
+		      bar:
+		        type: string
+		    required:
+		      - baz
```

Comment:
The error message here is `object has missing required properties %s`, where `%s` is the name of a property which is listed in the `required` list, but does not exist.  
To me something like `The referenced property %s does not exist`  or `property %s is not defined` would be more informative. 
@tedepstein , do you have suggestions for this case?